lib/puppet/settings.rb

```diff
@@ -839,20 +779,20 @@ def set_by_cli?(param)
     !@value_sets[:cli].lookup(param).nil?
   end
 
-  def set_value(param, value, type, options = {})
-    Puppet.deprecation_warning("Puppet.settings.set_value is deprecated. Use Puppet[]= instead.")
+  # Patches the value for a param in a section.
+  # This method is required to support the use case of unifying --dns-alt-names and
+  # --dns_alt_names in the certificate face. Ideally this should be cleaned up.
+  # See PUP-3684 for more information.
+  # For regular use of setting a value, the method `[]=` should be used.
+  # @api private
+  #
+  def patch_value(param, value, type, options = {})
```

Comment:
excellent question, I just renamed the method and did not notice the default dead options. Will remove it.